android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/RecoveryPhraseFragment.java

```diff
@@ -93,4 +98,15 @@ private void setupRecoveryPhraseRecyclerView(View view) {
         recoveryPhraseAdapter.setRecoveryPhraseList(recoveryPhrases);
         recyclerView.setAdapter(recoveryPhraseAdapter);
     }
+
+    @Override
+    public void onIntentCompleted(int resultCode, Intent results) {
+        if (resultCode != Activity.RESULT_OK || results == null || results.getData() == null) {
+            Utils.onFileNotSelected(getActivity());
+            return;
+        }
+
+        Uri uri = results.getData();
+        Utils.writeTextToFile(getActivity(), uri, Utils.getRecoveryPhraseFromList(recoveryPhrases));
```

Comment:
I believe that's executed on UI thread. We cannot save files in UI thread, it has to be done via `java.util.concurrent` package. That's an example:
```
    ExecutorService executor = Executors.newSingleThreadExecutor();
    Handler handler = new Handler(Looper.getMainLooper());

    executor.execute(() -> {
        //Background work here
        handler.post(() -> {
            //UI Thread work here
        });
    });
```
I don't think you need any UI thread work after file saved, so `post` part could be ignored.